hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -122,6 +123,15 @@ public DatanodeDetails getDatanodeDetails() {
     public long getStartTime() {
       return startTime;
     }
+
+    public Map<String, List<ContainerID>> getContainersReplicatedOnNode() {
+      return containersReplicatedOnNode;
+    }
+
+    public void setContainersReplicatedOnNode(List<ContainerID> underReplicated, List<ContainerID> unClosed) {
+      this.containersReplicatedOnNode.put("UnderReplicated", ImmutableList.copyOf(underReplicated));
```

Comment:
I don't think we need to make a copy of the lists here. The list are local variable so nothing can change them after we exit the method, so making a copy just adds expense. It would be find to wrap them in an Immutable list however.

Reply:
Thank you for the review! I have changed this to Collections.unmodifiableList(containerList)